doc/guides/dts/macros.rst

```diff
@@ -245,112 +336,117 @@ on the node, and to 0 otherwise:
 
    #define DT_<node>_FOO 0/1
 
-For non-boolean types the property macros are not generated if the ``category``
-is ``optional`` and the property is not present in the devicetree source.
+For non-boolean types the property macros are not generated if the binding's
+``category`` is ``optional`` and the property is not present in the devicetree
+source.
+
+.. _phandle-array-macros:
+
+Properties with type ``phandle-array``
+======================================
 
 The generation for properties with type ``phandle-array`` is the most complex.
 To understand it, it is a good idea to first go through the documentation for
 ``phandle-array`` in :ref:`dt-bindings`.
 
 Take the following devicetree nodes and binding contents as an example:
 
-.. code-block:: none
+.. code-block:: DTS
    :caption: Devicetree nodes for PWM controllers
 
    pwm_ctrl_0: pwm-controller-0 {
         compatible = "vendor,pwm-controller";
         label = "pwm-0";
         #pwm-cells = <2>;
-        ...
+        /* ... */
    };
 
    pwm_ctrl_1: pwm-controller-1 {
         compatible = "vendor,pwm-controller";
         label = "pwm-1";
         #pwm-cells = <2>;
-        ...
+        /* ... */
    };
 
 .. code-block:: yaml
-   :caption: ``pwm-cells`` declaration in binding for ``vendor-pwm-controller``
+   :caption: ``pwm-cells`` declaration in binding for ``vendor,pwm-controller``
 
    pwm-cells:
        - channel
        - period
 
 Assume the property assignment looks like this:
 
-.. code-block:: none
+.. code-block:: DTS
 
    pwm-user@0 {
-           compatible = "vendor,foo";
+           compatible = "vendor,device";
            status = "okay";
            reg = <0 1024>;
-           pwms = <&pwm_ctrl_0 1 10
-                   &pwm_ctrl_1 2 20>;
+           pwms = <&pwm_ctrl_0 1 10>, <&pwm_ctrl_1 2 20>;
            pwm-names = "first", "second";
    };
 
-These macros then get generated:
+These macros then get generated.
 
 .. code-block:: none
 
-   #define DT_VENDOR_FOO_0_PWMS_CONTROLLER_0    "PWM_0"
```

Comment:
Was it on purpose to replace VENDOR_FOO with <node>?  If so I think we need to use <node> in the text below

Reply:
> Was it on purpose to replace VENDOR_FOO with ? If so I think we need to use in the text below

Looks like I missed you on your vacation -- yes, this was on purpose, but I also did replace it in the text below. Maybe you're looking at the interrupt related macros in a separate section? I can replace those too.